### PR TITLE
[cssom] Add tests for `CSSGroupingRule#insertRule`

### PR DESCRIPTION
**Pull request by jugglinmike**
These tests are somewhat contentious due an inconsistency in the CSS
specification:

https://github.com/w3c/csswg-drafts/issues/3528

### CODE REVIEW

**Comment by gsnedders** [comment]
(Needs a rebase to pass CI, due to #17286. Sorry!)

**Comment by jugglinmike** [comment]
No problem. Rebased!

File: css/cssom/CSSGroupingRule-insertRule.html [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        assert_equals(groupingRule.cssRules.length, 1);
+        assert_equals(groupingRule.cssRules[0].cssText, first);
+    }, 'CSS parsing error');
+
+    test(function (t) {
+        var groupingRule = create(t);
+        var first = groupingRule.cssRules[0].cssText;
+
+        assert_throws('HierarchyRequestError', function() {
+            groupingRule.insertRule('@import url ("foo.css");', 0);
```
Comment (zcorpan) [inline css/cssom/CSSGroupingRule-insertRule.html]: There shouldn't be a space between `url` and `(`

**Review by zcorpan — CHANGES_REQUESTED** [review]
LGTM % 1 bug :)

**Comment by jugglinmike** [comment]
Thanks for your eagle eye, @zcorpan

**Review by zcorpan — APPROVED** [review]

**Comment by zcorpan** [comment]
Spec PR https://github.com/w3c/csswg-drafts/pull/4057

**Comment by jugglinmike** [comment]
@zcorpan noted that the two definitions differ in their use of `[SameObject]` on the `cssRules` property. Since the definition I'm proposing we persist specifies that, I've added a test to ensure that it's honored.

File: css/cssom/CSSGroupingRule-cssRules.html [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+        assert_equals(groupingRule.cssRules.wptMarker, 'wpt');
+
+        try {
+            groupingRule.deleteRule('.foo {}', 0);
+            groupingRule.deleteRule('.bar {}', 0);
+            groupingRule.deleteRule('.baz {}', 0);
+        } catch (err) {}
+
+        assert_equals(groupingRule.cssRules.wptMarker, 'wpt');
```
Comment (zcorpan) [inline css/cssom/CSSGroupingRule-cssRules.html]: This test could pass if the returned object is a new structured clone each time. Add a straight `assert_equals` on the object directly, too.

Comment (jugglinmike) [reply css/cssom/CSSGroupingRule-cssRules.html]: I based this off of [existing CSSOM tests](https://github.com/web-platform-tests/wpt/blob/13b3630dfbea2e842b45a44c523077d0c9d1b08e/css/cssom/CSSStyleSheet.html). Asserting object equality will require creating additional references, so I assumed this wasn't done as a way to promote garbage collection.

Comment (zcorpan) [reply css/cssom/CSSGroupingRule-cssRules.html]: Ah, yeah, storing a reference would defeat the purpose of testing the effect of GC. But as-is it seems extremely unlikely for a GC to happen in the middle of running this test...

That said, I think this is good enough to merge.

**Comment by jugglinmike** [comment]
Thanks for the review, @zcorpan. [The spec change was accepted today](https://github.com/w3c/csswg-drafts/pull/4029), so I'll merge this myself.

**Pull request merged**